mkosi/__init__.py

```diff
@@ -1187,18 +1207,31 @@ def extract_kernel_image_initrd(state: MkosiState) -> None:
         return
 
     kimgabs = None
-    initrd = None
+    initrds = []
+    kmods_initrd = None
 
     for kver, kimg in gen_kernel_images(state):
         kimgabs = state.root / kimg
-        initrd = initrd_path(state, kver)
+        initrds = initrd_paths(state, kver)
+        kmods_initrd = kmods_initrd_path(state, kver)
 
     if kimgabs is None:
         return
-    assert initrd is not None
+    assert initrds
 
     copy_file(kimgabs, state.staging / state.config.output_split_kernel_image.name)
-    copy_file(initrd, state.staging / state.config.output_split_initrd.name)
+
+    if len(initrds) == 1:
+        copy_file(initrds[0], state.staging / state.config.output_split_initrd.name)
+    else:
+        for i, initrd in enumerate(initrds):
+            copy_file(initrd, state.staging / state.config.output_split_initrds(i).name)
```

Comment:
I'm not sure I understand why there's a need to split this and have both `output_split_initrd` and `output_split_initrds`, which might get confusing.

Reply:
I did this to avoid a breaking change if people rely on the existing filename. I keep the existing name (without index) if there is only one initrd; otherwise I append an index.

Reply:
We should handle this a bit differently. Instead of outputting multiple initrds, we should output the combined initrd section that's generated by ukify. You can do that by using llvm-objcopy or objcopy with the --dump-section command.

`objcopy --dump-section .initrd=<output_split_initrd`. We should prefer using llvm-objcopy if it is available and fallback to objcopy otherwise.

Reply:
I have the following error when I use `llvm-copy` to extract the initrd:
```
llvm-objcopy: error: option is not supported for COFF
```

Reply:
This actually shouldn't be necessary anymore after we merge https://github.com/systemd/mkosi/pull/1301